crates/sui-json-rpc/src/api/read.rs

```diff
@@ -31,12 +31,27 @@ pub trait ReadApi {
         query: Option<SuiObjectResponseQuery>,
         /// An optional paging cursor. If provided, the query will start from the next item after the specified cursor. Default to start from the first item if not specified.
         cursor: Option<ObjectID>,
-        /// Max number of items returned per page, default to [MAX_GET_OWNED_OBJECT_SIZE] if not specified.
+        /// Max number of items returned per page, default to [QUERY_MAX_RESULT_LIMIT_OBJECTS] if not specified.
         limit: Option<usize>,
         /// If not specified, objects may be created or deleted across pagination requests. This parameter is only supported when the sui-indexer instance is running.
         at_checkpoint: Option<CheckpointId>,
     ) -> RpcResult<ObjectsPage>;
 
+    #[method(name = "queryObjects")]
```

Comment:
This method should only be available on indexer, just like the epoch API, since we will remove index from fullnode soon: https://github.com/MystenLabs/sui/pull/9588/files#diff-315f1117edaf2938a84da8a3155cead9d7ea59d9334a8e7dd0151240bb9eb4d9

Reply:
Should I wait on @patrickkuo to make this change? The extended layer isn't added yet.

Reply:
yep, at least the `All` and `Any` can only be done on indexer imo.